core/src/main/java/com/ibm/wala/ipa/cha/ClassHierarchy.java

```diff
@@ -706,6 +707,111 @@ private void recursiveStringify(Node n, StringBuilder buffer) {
     }
   }
 
+  /*
+   * Puts ClassHierarchy into a JSON readable variable that maps the class to a list of its
+   * subclasses
+   */
+  public Object toJson() {
+    /*
+     * initialize variables (subclass and dag) to store important values like hashmap of each
+     * iteration and a list of all maps of class to subclasses
+     */
+    HashMap<String, Set<String>> subclass = new HashMap<>();
+    Iterator<Node> children = root.getChildren();
+    /*
+     * Goes through all subclasses of the root class, and adds the subclass hashmap to the
+     * list of all subclass hashmaps
+     */
+    while (children.hasNext()) {
+      Node temp = children.next();
+      helperToJson(temp, subclass);
+    }
+    /*
+     * Removes unnecesarry parts from name of the class
+     */
+    String key = nodeToString(root);
+    /*
+     * inserting the root class to its subclasses into the main hashmap while removing
+     * unnecessary substrings from the class name
+     */
+    Iterator<Node> root_children = root.getChildren();
```

Comment:
Note camel case again; this should be `rootChildren`.  Please update all local variable names appropriately